vortex-array/src/arrays/varbin/compute/take.rs

```diff
@@ -62,19 +71,34 @@ fn take_nullable<I: NativePType, O: NativePType + PrimInt>(
     offsets: &[O],
     data: &[u8],
     indices: &[I],
-    null_buffer: NullBuffer,
+    data_validity: Mask,
+    indices_validity: Mask,
 ) -> VarBinArray {
+    let data_null_buffer = data_validity
```

Comment:
There's a bit of legacy cruft here. You can replace usages of NullBuffer with using the mask directly